bsmetadata/preprocessing_tools/html_parser/filters_and_cleaners.py

```diff
@@ -0,0 +1,459 @@
+                previous.tail = node.tail
+            elif (
+                previous.text
+                and not previous.text.endswith(PLAIN_TEXT_SEPARATOR)
+                and not node.tail.startswith(PLAIN_TEXT_SEPARATOR)
+            ):
+                previous.text = previous.text + PLAIN_TEXT_SEPARATOR + node.tail
+            elif (
+                previous.text
+                and previous.text.endswith(PLAIN_TEXT_SEPARATOR)
+                and node.tail.startswith(PLAIN_TEXT_SEPARATOR)
+            ):
+                # Don't accumulate too much spaces
+                previous.text = previous.text[: -len(PLAIN_TEXT_SEPARATOR)] + node.tail
+            elif (
+                previous.tail
+                and not previous.tail.endswith(PLAIN_TEXT_SEPARATOR)
+                and not node.tail.startswith(PLAIN_TEXT_SEPARATOR)
+            ):
+                previous.tail = previous.tail + PLAIN_TEXT_SEPARATOR + node.tail
+            else:
+                previous.tail = previous.tail + node.tail
+        else:  # The parent get the tail as text
+            parent = node.getparent()
+            if parent.text is None:
+                parent.text = node.tail
+            elif not parent.text.endswith(PLAIN_TEXT_SEPARATOR) and not node.tail.startswith(PLAIN_TEXT_SEPARATOR):
+                parent.text = parent.text + PLAIN_TEXT_SEPARATOR + node.tail
+            elif parent.text.endswith(PLAIN_TEXT_SEPARATOR) and node.tail.startswith(PLAIN_TEXT_SEPARATOR):
+                # Don't accumulate too much spaces
+                parent.text = parent.text[: -len(PLAIN_TEXT_SEPARATOR)] + node.tail
+            else:
+                parent.text = parent.text + node.tail
+
+
+class TextAndMetadataCleaner:
+    def __init__(
+        self,
+        html_str,
+        tags_to_remove_with_content: Optional[List[TagToRemoveWithContent]] = None,
+        tags_to_remove_alone: Optional[List[TagToRemove]] = None,
+        attrs_to_keep: Optional[List[str]] = None,
+        start_parsing_at_tag: Optional[str] = "body",
+        consecutive_tags_to_fold: Optional[List[str]] = None,
+        convert_br_tag_to_breaking_line: Optional[bool] = False,
+        txt_max_chr_len_alone: float = -float("inf"),
+        txt_min_chr_len_alone: float = -float("inf"),
+        tags_exceptions_to_txt_max_min_chr_len_alone: List[str] = None,
+        txt_max_chr_len_with_content: float = -float("inf"),
+        txt_min_chr_len_with_content: float = -float("inf"),
+        tags_exceptions_to_txt_max_min_chr_len_with_content: List[str] = None,
+    ):
+        self.html_str = html_str
+        self.tags_to_remove_with_content = tags_to_remove_with_content
+        self.tags_to_remove_alone = tags_to_remove_alone
+        self.attrs_to_keep = attrs_to_keep
+        self.start_parsing_at_tag = start_parsing_at_tag
+        self.convert_br_tag_to_breaking_line = convert_br_tag_to_breaking_line
+
+        self.tags_to_remove_alone = (
```

Comment:
Nitpick: If you'd change this to 
```python
if self.tags_to_remove_alone is None:
    self.tags_to_remove_alone = []
self.tags_to_remove_alone = [
    TagToRemove(...),
    ...
]
```
the list of fake tags doesn't need to be repeated.

Reply:
That's a good suggestion, thanks! I just committed it :slightly_smiling_face: